components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/AbstractUserInfoResponseBuilder.java

```diff
@@ -61,7 +61,8 @@ public abstract class AbstractUserInfoResponseBuilder implements UserInfoRespons
     public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
             throws UserInfoEndpointException, OAuthSystemException {
 
-        String clientId = getClientId(OAuth2Util.getAccessTokenIdentifier(tokenResponse));
+        AccessTokenDO accessTokenDO = OAuth2Util.getAccessTokenDO(tokenResponse);
+        String clientId = accessTokenDO.getConsumerKey();
```

Comment:
Since the getAccessTokenDO can return a null response  accessTokenDO.getConsumerKey() has the possibility of NPE.

Lets update the method getAccessTokenDO to return a java optional

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2176/commits/ae287fa0b87cd0e48f8fd62b035ca1a6e7fca706